pkg/master/reconcilers/none.go

```diff
@@ -0,0 +1,38 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package reconcilers a noop based reconciler
+package reconcilers
+
+import (
+	"net"
+
+	"k8s.io/kubernetes/pkg/api"
+)
+
+// NoneEndpointReconciler allows for the endpoint reconciler to be disabled
+type NoneEndpointReconciler struct{}
```

Comment:
Is there any reason this type is exported?

Same goes for other reconciler structs.

Reply:
I know that OpenShift often configures and starts up things like the master, and it needs full access to the types. cc @deads2k @liggitt

Reply:
Why though? At least for `None` it can certainly still call `NewNoneEndpointReconciler`. The type adds nothing here since the constructor is still exported.

Reply:
Sure, if there's a constructor, can just export the interface & the constructors.